OpenRA.Game/Graphics/SpriteCache.cs

```diff
@@ -108,26 +108,27 @@ public void LoadReservations(ModData modData)
 						else
 						{
 							resolvedFrames[token] = null;
-							missingFiles[token] = (filename, r.Location);
+							missingFiles[token] = (filename, rf.Location);
 						}
 					}
 
-					if (spriteReservations.TryGetValue(token, out r))
+					if (spriteReservations.TryGetValue(token, out var rs))
 					{
 						if (loadedFrames != null)
 						{
 							var resolved = new Sprite[loadedFrames.Length];
-							var frames = r.Frames ?? Enumerable.Range(0, loadedFrames.Length);
+							var frames = rs.Frames ?? Enumerable.Range(0, loadedFrames.Length);
+							var di = rs.Premultiplied ? loadedFrames.Length : 0;
 							foreach (var i in frames)
-								resolved[i] = spriteCache.GetOrAdd(i,
-									f => SheetBuilders[SheetBuilder.FrameTypeToSheetType(loadedFrames[f].Type)].Add(loadedFrames[f]));
+								resolved[i] = spriteCache.GetOrAdd(i + di,
+									f => SheetBuilders[SheetBuilder.FrameTypeToSheetType(loadedFrames[f - di].Type)].Add(loadedFrames[f - di], rs.Premultiplied));
```

Comment:
the `di` changes don't really make much sense to me. Could this get an in-code explanation?